### PR TITLE
feat: Refactored icon component

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This pull request presents an improved version of the icon component, including enhancements in theme customisation, keyboard accessibility, and API usability.

Fixes [#19332](https://github.com/appsmithorg/appsmith/issues/19332)

Depends on #318

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2152

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### QA activity:
- [x] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Icon/Icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+
+import { Icon as IconComponent } from "./Icon";
+
+export default {
+  title: "Design System V2/Icon",
```
Comment (tanvibhakta) [inline packages/design-system/src/Icon/Icon.stories.tsx:7]: Design System

File: packages/design-system/src/Icon/Icon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+
+import { Icon as IconComponent } from "./Icon";
```
Comment (tanvibhakta) [inline packages/design-system/src/Icon/Icon.stories.tsx:4]: Why are we importing this as IconComponent? Let us rename the story using the `.storyname` method instead.

Comment (albinAppsmith) [reply packages/design-system/src/Icon/Icon.stories.tsx:4]: .storyname?

**Review by tanvibhakta — APPROVED** [review]

**Comment by shadabbuchh** [comment]
This PR has been tested. Working as expected.

**Comment by shadabbuchh** [comment]
As per our discussion, mentioning issues related to wrong developer input in the Control fields:

1. Entering irrelevant input in the `size` & `colour` controls does not throw any error.
2. Simplify the error logs that are displayed on entering irrelevant input in the `name ` control.

**Pull request merged**